src/allotropy/parsers/cytiva_biacore_t200_evaluation/cytiva_biacore_t200_evaluation_data_creator.py

```diff
@@ -0,0 +1,513 @@
+        if kinetic_result.fit_quality.chi2_value is not None
+        else None
+    )
+
+
+def _create_report_points_from_cycle_data(
+    rp_df: pd.DataFrame | None,
+    flow_cell_id: str,
+    cycle_number: int,
+    display_flow_cell_id: str | None = None,
+) -> list[ReportPoint] | None:
+    """Create ReportPoint objects from cycle report point data, filtered by flow cell."""
+    if rp_df is None or rp_df.empty:
+        return None
+
+    report_points: list[ReportPoint] = []
+
+    filtered_df = rp_df
+    if "Flow Cell Number" in rp_df.columns or "flow_cell" in rp_df.columns:
+        # Try to filter by flow cell
+        fc_col = (
+            "Flow Cell Number" if "Flow Cell Number" in rp_df.columns else "flow_cell"
+        )
+        # Convert flow_cell_id to match the format in the DataFrame
+        try:
+            fc_filter_value = int(flow_cell_id)
+            filtered_df = rp_df[rp_df[fc_col] == fc_filter_value]
+        except (ValueError, KeyError):
+            # If filtering fails, use all data (fallback)
+            filtered_df = rp_df
+
+    # Map column names based on specification:
+    # column1 = Time setting, column3 = Relative resonance, column4 = Identifier role, column5 = Absolute resonance
+    for _idx, row in filtered_df.iterrows():
+        try:
+            # Extract values from the expected columns
+            time_setting = try_float_or_none(
+                str(row.get("column1") or row.get("Time") or 0.0)
+            )
+            relative_resonance = try_float_or_none(
+                str(row.get("column3") or row.get("Relative") or 0.0)
+            )
+            identifier_role = str(row.get("column4") or row.get("Role") or "baseline")
+            absolute_resonance = try_float_or_none(
+                str(row.get("column5") or row.get("Absolute") or 0.0)
+            )
+
+            # Use display flow cell ID for identifiers, fallback to base flow cell ID
+            fc_id_for_display = display_flow_cell_id or flow_cell_id
+            # Generate a unique identifier for this report point (include cycle number)
+            report_point_id = f"CYTIVA_BIACORE_T200_EVALUATION_RP_C{cycle_number}_FC{fc_id_for_display}_{random_uuid_str()}"
+
+            report_points.append(
+                ReportPoint(
+                    identifier=report_point_id,
+                    identifier_role=identifier_role,
+                    absolute_resonance=absolute_resonance or 0.0,
+                    time_setting=time_setting or 0.0,
+                    relative_resonance=relative_resonance,
+                    custom_info={"window": {"value": 5.0, "unit": "s"}},
+                )
+            )
+        except Exception:  # noqa: S112
+            # Skip malformed rows - acceptable for data parsing
+            continue
+
+    return report_points if report_points else None
+
+
+def _create_measurements_for_cycle(_: Data, cycle: CycleData) -> list[Measurement]:
```

Comment:
Please give this a variable name, since you are using it.

```suggestion
def _create_measurements_for_cycle(data: Data, cycle: CycleData) -> list[Measurement]:
```